### DISCUSSION TITLE
Maybe crazy idea?

### DISCUSSION THREAD

**Discussion by sergiobellini** [corrados/edrumulus#134] [Ideas]
Hi Corrados, congratulations on your work.

As a Hobby I am just recently approaching the world of microcontrollers, and I am a developer. I would like to ask you a question, considering all the study you have already done on ADCs and triggering with piezos... do you think it is feasible to make a radio transmission system (not necessarily Wifi or Bluetooth, maybe better analog transmission with proprietary protocol) between an ESP32/Teensy and an external module? The external module might contain an ADC, or it might not. How much would it affect the latency that you were able to optimize instead? And if this is theoretically feasible, could it be extended for a higher number of external modules? (8/10)

Thank you for your attention.

**Comment by corrados** [comment]
Roland/DW does wireless data transfer from the pads to the module. So, in theory it is possible with good latency. If you want to use Bluetooth, I think the latency will be too high. With Wifi, I don't know. If you want to use Wifi or Bluetooth in the ESP32 together with the Edrumulus code, that is not possible. Edrumulus needs all CPU cores and you get a segmentation fault if you enable either Bluetooth or Wifi.

**Reply by sergiobellini** [reply]
Hi Corrados, thank you for your response ... I will go and study Roland's system, where can I find information? What exactly are you referring to?

**Reply by corrados** [reply]
E.g., https://www.youtube.com/watch?v=skTLAcs4gQs&t=16s

**Reply by sergiobellini** [reply]
Wow, I didn't know about this device! Thank you.

**Comment by sergiobellini** [comment]
Hi Corrados, I would like to start building my breadboard for connecting piezos... currently I have a simple piezo on a small Millenium splash (like this [https://www.flymusic.ro/pad-uri-pt-toba- electronica/19321-millenium-mps-300-mono-cymbal-pad.html](url) ).
I'm following the prototype2 pattern, how do I start, is that ok? I currently don't have BAT85 diodes, but 1N4148 ones, do you think they're fine anyway?
Building it (with the 1N4148 I have) I see with the oscilloscope a signal that starts from 1.60v and reaches a maximum of 3.6v hitting the plate with maximum force, is the range correct?
Thanks so much!

**Reply by corrados** [reply]
> I currently don't have BAT85 diodes, but 1N4148 ones, do you think they're fine anyway?

Yes. On some of my prototypes, I have also used normal diodes and it works fine for me.

> Building it (with the 1N4148 I have) I see with the oscilloscope a signal that starts from 1.60v and reaches a maximum of 3.6v hitting the plate with maximum force, is the range correct?

Yes, these numbers look ok to me.

**Reply by sergiobellini** [reply]
Perfect, thank you very much, I'll get to work :)

I saw the presence of other prototypes besides 2, and in my case I was interested in 6, i.e. a piezo board with MIDI OUT interface. I don't think I found a PDF of the electrical diagram of prototype 6 on the repository, did I see wrong?

**Reply by corrados** [reply]
Unfortunately, prototype 6 hardware has bugs (e.g., if the board is attached it cannot be flashed). I do not recommend using this prototype.

**Comment by sergiobellini** [comment]
Hi Corrados, before writing I checked everything many times, but I can't figure it out. I simply wanted to connect a piezo to your signal conditioning circuit, and read it with my oscilloscope... if the schematic is correct, what happens to me is I read a constant voltage of about 1.6v, but hitting the piezo does not notice practically no change... did I make a mistake with the connections?

![Screenshot 2024-03-08 215839](https://github.com/corrados/edrumulus/assets/85103743/039c3815-569d-40b4-9e3a-45be34d1ca11)

**Reply by corrados** [reply]
Your schematics look ok. I have no idea why you are not seeing anything on your oscilloscope. The only thing I can say is that the schematics in combination with the ESP32 work for me...

**Reply by sergiobellini** [reply]
ok, then I'll try to log the analog read :) I'll update you

**Reply by sergiobellini** [reply]
Unfortunately I didn't have a 22 resistor and I was mounting a 20k + a 2k together, but in an incorrect way :( how stupid. The reading is correct! Thank you very much!

**Comment by sergiobellini** [comment]
Hi Corrados, sorry if I continue on this thread, but I didn't want to open another, perhaps useless one.
I built the circuit for reading the piezos on a breadboard and everything works correctly, out of curiosity I wanted to understand (or what document I need to read) why your project is based on reading, on the microcontroller, a signal that "starts" from that voltage obtained from the voltage divider, and which obviously (due to the behavior of the piezos) then develops a shape that provides higher values than that reference, as well as lower values. Does the software algorithm also take these variations into account?

**Reply by corrados** [reply]
Yes, it does. I do a lot of FIR filtering in my algorithms and they need the signal to be original and not clipped.
